### PR TITLE
Scheduler#onNavigate() now provides NavigateEvent as parameter containing old-view and new-view; fixes #233

### LINKED ISSUE #233: Scheduler#onNavigate() / is SchedulerViewType the new or previous view?
Hi Sebastien,
what is the meaning of SchedulerViewType on `onNavigate(..., SchedulerViewType view)`.

The new view or the previous view?

Currently it seems it is the previous view.

kind regards
Patrick

### PR DESCRIPTION
**Pull request by Patrick1701**
Hi Sebastien,
please have a review, and if ok, please add it to your next release.

kind regards
Patrick

### CODE REVIEW

**Comment by sebfz1** [comment]
Will try to do it this week, thanks Patrick ! :)

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/ISchedulerListener.java [hunk 1]
```diff
@@ -60,9 +61,9 @@
 	 * Triggered when the user changes the selected date, or view of the scheduler
 	 * 
 	 * @param target the {@link AjaxRequestTarget}
-	 * @param view the {@link SchedulerViewType}
+	 * @param navigateEvent the {@link NavigateEvent}
 	 */
-	void onNavigate(AjaxRequestTarget target, SchedulerViewType view);
+	void onNavigate(AjaxRequestTarget target, NavigateEvent navigateEvent);
```
Comment (sebfz1) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/ISchedulerListener.java]: `event`

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/Scheduler.java [hunk 2]
```diff
@@ -340,7 +341,7 @@ public void onEdit(AjaxRequestTarget target, SchedulerEvent event, SchedulerView
 	}
 
 	@Override
-	public void onNavigate(AjaxRequestTarget target, SchedulerViewType view)
+	public void onNavigate(AjaxRequestTarget target, NavigateEvent navigateEvent)
```
Comment (sebfz1) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/Scheduler.java]: `event`

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/SchedulerAdapter.java [hunk 3]
```diff
@@ -43,7 +44,7 @@ public void onEdit(AjaxRequestTarget target, SchedulerEvent event, SchedulerView
 	}
 
 	@Override
-	public void onNavigate(AjaxRequestTarget target, SchedulerViewType view)
+	public void onNavigate(AjaxRequestTarget target, NavigateEvent navigateEvent)
```
Comment (sebfz1) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/SchedulerAdapter.java]: `event`

File: wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/SchedulerBehavior.java [hunk 4]
```diff
@@ -217,7 +216,7 @@ public void onAjax(AjaxRequestTarget target, JQueryEvent event)
 	{
 		if (event instanceof NavigateEvent)
 		{
-			this.listener.onNavigate(target, ((NavigateEvent) event).getView());
+			this.listener.onNavigate(target, ((NavigateEvent) event));
```
Comment (sebfz1) [inline wicket-kendo-ui/src/main/java/com/googlecode/wicket/kendo/ui/scheduler/SchedulerBehavior.java]: extra parenthesis

**Comment by sebfz1** [comment]
HI Patrick,

Seems fine ! :)
I just did some minor comments...

Thanks & best regards,
Sebastien

**Comment by Patrick1701** [comment]
Ok, ähm, any todos for me, or just notes for you?

**Comment by sebfz1** [comment]
Todo for you if you can :)

**Comment by Patrick1701** [comment]
```event``` means... just rename?

```extra parenthesis``` means too much?

;-)

**Comment by sebfz1** [comment]
Yes and yes :)

**Comment by Patrick1701** [comment]
Ok, concerning the parameter naming...
I didn't use ```event```because, I thought the naming also ment it is an ```SchedulerEvent``` in sense of appointment-event and not a technical event. So I thought ```navigateEvent``` would be clearer.
So, just to be sure, you would like to name it ```event```, right?

**Comment by sebfz1** [comment]
Hi Patrick, your point of view is 100℅ correct, but due to the coding style
in the project, I still prefer event as an argument name :)

Thanks a lot,
Sebastien

**Comment by Patrick1701** [comment]
Hi Sebastien, done.

Best regards
Patrick

**Pull request merged**